### PR TITLE
Tests function on Mac

### PR DESCRIPTION
**Pull request by duncanbarth**
I found that the tests for cshatag doesn't work in MacOS - but the tool itself will compile and function. 

I hacked in some conditionals to the test script so it will also run under MacOS. Its pretty ugly - and makes some assumptions that may not always be correct (the main one being that if set/getfattr exists we're on linux, otherwise if xattr exists assume we're on a Mac). It will however run all tests on a Mac without any packages outside of the xcode cli tools and go.

I'm totally open to suggestions for improvement here!

### CODE REVIEW

File: tests/run_tests.sh [hunk 1]
```diff
@@ -98,7 +135,14 @@ if [[ $RES -eq 0 ]]; then
 	echo "should have returned an error code, but returned 0"
 	exit 1
 fi
-diff -u 5.expected 5.err
+
+# MacOS returns ENOATTR instead of ENODATA on the remove
+if [ "$ATTRVER" = "linux" ]
```
Comment (rfjakob) [inline tests/run_tests.sh:140]: if [[ $(uname) == Darwin ]] ; then
diff -u 5.expected.mac 5.err
else
...
fi

File: tests/run_tests.sh [hunk 2]
```diff
@@ -12,8 +12,26 @@ function cleanup {
...
+else
+	# no getfattr, see if we have xattr
+	if command -v xattr > /dev/null 2>&1
+	then
+		ATTRVER="macos"
+	else
+		echo "error: no getfattr or xattr available"
+		exit 1
+	fi
+fi
```
Comment (rfjakob) [inline tests/run_tests.sh:34]: Just delete the "have gefattr / setfattr" check without replacement, it's not worth the hassle

**Review by rfjakob — CHANGES_REQUESTED** [review]
Hi, thanks for the PR! Having the tests work on MacOS would be nice.

I suggest to introduce bash functions `set_sha256`, `set_ts`, `get_sha256`. They would use setfattr/getfattr if available, then xattr, then fail if neither is available.

A few more comments inline.

**Comment by duncanbarth** [comment]
I probably should have squashed those commits, but in any case, I've implemented your suggested changes.

**Comment by rfjakob** [comment]
Merged as 997c309, thanks!

**Pull request closed without merging**